prototype2/gdgem/NMXConfig.h

```diff
@@ -50,11 +51,14 @@ struct EventFilter {
 };
 
 struct NMXConfig {
-  NMXConfig() {}
-  NMXConfig(std::string jsonfile);
+  NMXConfig() { }
+  NMXConfig(std::string configfile, std::string calibrationfile);
 
   std::string builder_type{"VMM2"};
 
+  // VMM calibration
+  CalibrationFile * calfile;
```

Comment:
I would strongly prefer some smart-pointer use here.

Reply:
I am not smart enough to use smart pointers. Whenever I try I end up wasting 
a ton of time trying to get it to work :-)

Reply:
Ok, very quick lesson:
* declare it as `std::shared_ptr<CalibrationFile> calfile` no need for a default value
* when assigning do `calfile = std::make_shared<CalibrationFile>(par1, par2)` with parameters as if you were using a constructor of your choice
* you can check for validity like `if (calfile)`
* get undelrying pointer with `calfile.get()`
* use it as any dumb pointer in every other respect, i.e. `->` to access members
* pass `std::shared_ptr<CalibrationFile>` around by value
* if you don't intend to pass it around, use `unique_ptr` instead and all of the above still holds true